index.js

```diff
@@ -45,6 +41,23 @@ module.exports = function (opts) {
 			cb(null, file);
 		};
 
+		if (file.isStream()) {
+			if (opts.gzip) {
+				file.contents.pipe(gzipSize.stream())
+					.on('error', finish)
+					.on('end', function () {
+						finish(null, this.gzipSize);
+					});
+			} else {
+				file.contents.pipe(new StreamCounter())
+					.on('error', finish)
+					.on('finish', function () {
+						finish(null, this.bytes);
+					});
+			}
+			return;
```

Comment:
`StreamCounter` returns a writeable stream, so those emit a `finish` event. `gzipSize.stream()` returns a [stream](https://github.com/Raynos/duplexer/blob/v0.1.1/index.js#L20) instance, created by duplexer, which only proxies the [`end`](https://github.com/Raynos/duplexer/blob/v0.1.1/index.js#L29) event of the [underlying readable stream](https://github.com/sindresorhus/gzip-size/blob/v2.1.0/index.js#L26).